libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -234,35 +235,26 @@ internal class MapboxTripSession(
 
             var triggerObserver = false
             if (tripStatus.navigationStatus.routeState != RouteState.INVALID) {
-                route?.let { route ->
-                    ifNonNull(route.legs()) { legs ->
-                        val currentLeg = legs[tripStatus.navigationStatus.legIndex]
-                        ifNonNull(currentLeg?.steps()) { steps ->
-                            val nativeBannerInstruction: BannerInstruction? =
-                                tripStatus.navigationStatus.bannerInstruction.let {
-                                    if (it == null &&
-                                        bannerInstructionEvent.latestBannerInstructions == null
-                                    ) {
-                                        // workaround for a remaining issues in
-                                        // github.com/mapbox/mapbox-navigation-native/issues/3466
-                                        // and
-                                        // github.com/mapbox/mapbox-navigation-android/issues/4727
-                                        MapboxNativeNavigatorImpl.getBannerInstruction(
-                                            status.stepIndex
-                                        )
-                                    } else {
-                                        it
-                                    }
-                                }
-                            val bannerInstructions: BannerInstructions? =
-                                nativeBannerInstruction?.mapToDirectionsApi()
-                            triggerObserver = bannerInstructionEvent.isOccurring(
-                                bannerInstructions,
-                                nativeBannerInstruction?.index
-                            )
+                val nativeBannerInstruction: BannerInstruction? =
+                    tripStatus.navigationStatus.bannerInstruction.let {
+                        if (it == null &&
+                            bannerInstructionEvent.latestBannerInstructions == null
+                        ) {
+                            // workaround for a remaining issues in
+                            // github.com/mapbox/mapbox-navigation-native/issues/3466
```

Comment:
This seems to be fixed by now.

Reply:
We have not migrated to async route setter yet.